addons/account/models/chart_template.py

```diff
@@ -82,11 +82,14 @@ def _get_chart_template_mapping(self, get_all=False):
                  name, country id, country name, module dependencies and parent template
         :rtype: dict[str, dict]
         """
+        # This function is called many times. Avoid doing a search every time by using the ORM's cache.
+        # We assume that the field is always computed for all the modules at once (by this function)
         field = self.env['ir.module.module']._fields['account_templates']
-        if self.env.cache.contains_field(field):
-            modules = self.env.cache.get_records(self.env['ir.module.module'], field)
-        else:
-            modules = self.env['ir.module.module'].search([])
+        modules = (
+            self.env.cache.get_records(self.env['ir.module.module'], field)
+            or self.env['ir.module.module'].search([])
```

Comment:
Is it not risky for performances since everytime `get_records` will not give any record, you will perform a `search`? I don't know what you mean by "many times" if it's called 10x or 1000x but in the worse case, you will perform n-search then.

Reply:
> I don't know what you mean by "many times"

Maybe a few hundred times, during the CoA instantiation, where these values are not invalidated.

> Is it not risky for performances since everytime `get_records` will not give any record, you will perform a `search`?
Basically, this should happen only once, until all the environments are reset, which doesn't happen often.